### PR TITLE
Update dependencies and fix BC accordingly

### PR DESCRIPTION
**Pull request by KentarouTakeda**
## Overview

This package has slightly older dependencies, making it a little more difficult to install on frameworks including Laravel and Symfony. (For example, currently, the `-w` option is required for `composer require` because `psr/http-message` needs to be downgraded during installation.)

To address these issues, I have expanded or updated the versions of some of the most frequently used packages.

* `psr/http-message`
  * Adopted @luiscunhaafricainternetgroup's pull request in #213. Thank you for your contribution.

* `league/uri`
  * Adopted @deegital's pull request in #205. Thank you for your contribution.

* Suppress errors caused by version changes in `league/uri`
  * The CI failure in #205 has been addressed. I have suppressed the errors reported there. Details are provided in the comments of the relevant code. This is due to the changes mentioned [here](https://github.com/thephpleague/uri/releases/tag/7.0.0).
    > `League\Uri\UriString` class. Class moved to the `uri-interfaces` package.

* `guzzlehttp/psr7`
   * Changed to `^2.0`.
   * `^2.0` contains breaking changes, so I have fixed all code related to them. Details of the fixes are described [here](https://github.com/guzzle/psr7?tab=readme-ov-file#upgrading-from-function-api). I simply replaced the method.

* *UPDATE: 2024-01-08 13:30 UTC* `psr/http-message `
  * Add support to 4.0 by hansott/psr7-cookies#17

## Additional Information

I previously created a pull request in #215 that significantly expanded the scope of CI. Although it has not been merged yet, I have tested this pull request with that commit in my repository.

In addition to the PHP versions supported by the current CI, we can confirm that the tests pass on recent PHP versions, including the latest version. The results can be viewed [here](https://github.com/KentarouTakeda/thephpleague-openapi-psr7-validator/actions/runs/7405574471).

## Note

~~Please note that even after importing this pull request, `psr/http-message` will not be upgraded to `2.0` due to other packages requiring version `1.0`.~~

~~I am also planning to submit a pull request to the repository of that package to address this issue.~~

*UPDATE: 2024-01-08 13:30 UTC*
That pull request is accepted and published as `4.0`, so I updated this pull request to follow it.

### CODE REVIEW

**Comment by KentarouTakeda** [comment]
> I am also planning to submit a pull request to the repository of that package to address this issue.

I sent it:
https://github.com/hansott/psr7-cookies/pull/17

**Comment by KentarouTakeda** [comment]
@scaytrase 
Hi, could you please check out these changes?

**Comment by KentarouTakeda** [comment]
The support for `hansott/psr7-cookies` has been completed, so we can now install the latest version of `psr/http-message` with the current code

The issue I mentioned at the beginning, about not being able to install this library in recent frameworks, is now **completely resolved**.

**Comment by KentarouTakeda** [comment]
@scaytrase 
I'm really sorry you're busy. Could you please take a look at this pull request?

If there's anything I can do to help, please let me know. I'll be happy to do it.

File: src/Schema/TypeFormats/StringURI.php [hunk 1]
```diff
@@ -12,6 +12,8 @@ class StringURI
     public function __invoke(string $value): bool
     {
         try {
+            // namespace 'League\Uri' is provided by multiple packages, but PHPStan does not support merging them
```
Comment (scaytrase) [inline src/Schema/TypeFormats/StringURI.php:15]: looks suspicious. can you give me more details, what packages bring similar namespaces?

File: src/Schema/TypeFormats/StringURI.php [hunk 2]
```diff
@@ -12,6 +12,8 @@ class StringURI
...
+            // @phpstan-ignore-next-line
```
Comment (scaytrase) [inline src/Schema/TypeFormats/StringURI.php:16]: I think this one could be reported to phpstan. really strange

**Review by scaytrase — APPROVED** [review]
👍 LGTM

**Comment by scaytrase** [comment]
Thanks @KentarouTakeda @deegital @luiscunhaafricainternetgroup !

**Comment by scaytrase** [comment]
https://github.com/thephpleague/openapi-psr7-validator/releases/tag/0.22

**Comment by KentarouTakeda** [comment]
Thanks a lot!

**Pull request merged**